production/ksonnet/loki/boltdb_shipper.libsonnet

```diff
@@ -0,0 +1,143 @@
+{
+  local pvc = $.core.v1.persistentVolumeClaim,
+  local volumeMount = $.core.v1.volumeMount,
+  local container = $.core.v1.container,
+  local statefulSet = $.apps.v1.statefulSet,
+  local service = $.core.v1.service,
+  local containerPort = $.core.v1.containerPort,
+
+  _config+:: {
+    // configurable pvc sizes
+    ingester_pvc_size: '5Gi',
+    querier_pvc_size: '10Gi',
+    compactor_pvc_size: '10Gi',
+
+    boltdb_shipper_shared_store: error 'must define boltdb_shipper_shared_store',
+    index_period_hours: 24,
+    loki+: {
+      chunk_store_config+: {
+        write_dedupe_cache_config:: {},
+      },
+      storage_config+: {
+        boltdb_shipper: {
+          shared_store: $._config.boltdb_shipper_shared_store,
+        },
+      },
+    },
+  },
+
```

Comment:
For cleanliness i ended up removing some options from the `table_manager_args` ... in my case i was moving from bigtable so i had something like 
```
  table_manager_args:: std.mergePatch(
    super.table_manager_args, {
      'bigtable.grpc-client-rate-limit': null,
      'bigtable.grpc-client-rate-limit-burst': null,
      'bigtable.backoff-on-ratelimits': null,
      'bigtable.table-cache.enabled': null,
    }
  ),
```

not sure if it make sense or , since there is no bigtable enabled, it does not really matter.

Reply:
Someone cloud be using both(moving to or away from bigtable) and would have to add an override again. It is safe to have those extra flags even if you are not using bigtable. We can take care of it by refactoring the jsonnet to include them only when bigtable is one of the stores but it is time-consuming so it would not happen anytime soon.

Reply:
Make sense. thanks